ddtrace/contrib/unittest/patch.py

```diff
@@ -0,0 +1,183 @@
+from ddtrace.ext import test
+from ddtrace.internal.ci_visibility import CIVisibility as _CIVisibility
+from ddtrace.internal.ci_visibility.constants import EVENT_TYPE as _EVENT_TYPE
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.vendor import wrapt
+
+
+# unittest default settings
+config._add(
+    "unittest",
+    dict(
+        _default_service="unittest",
+        operation_name=os.getenv("DD_UNITTEST_OPERATION_NAME", default="unittest.test"),
+    ),
+)
+
+
+def _set_tracer(tracer):
+    unittest._datadog_tracer = tracer
+
+
+def _store_span(item, span):
+    """Store span at `unittest` instance."""
+    item._datadog_span = span
+
+
+def _extract_span(item):
+    """Extract span from `unittest` instance."""
+    return getattr(item, "_datadog_span", None)
+
+
+def _extract_test_method_name(item):
+    """Extract test method name from `unittest` instance."""
+    return getattr(item, "_testMethodName", None)
+
+
+def _extract_suite_name_from_test_method(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_class_hierarchy_name(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_module_name_from_test_method(item):
+    return getattr(item, "__module__", None)
+
+
+def _extract_test_skip_reason(args):
+    return args[1]
+
+
+def _extract_test_file_name(item):
+    return os.path.basename(inspect.getfile(item.__class__))
+
+
+def is_unittest_support_enabled():
+    return unittest and getattr(unittest, "_datadog_patch", False) and _CIVisibility.enabled
+
+
+def _is_valid_result(instance, args):
+    return instance and type(instance) == unittest.runner.TextTestResult and args
+
+
+def patch():
+    """
+    Patch the instrumented methods from unittest
+    """
+    if (
+        not config._ci_visibility_unittest_enabled
+        or getattr(unittest, "_datadog_patch", False)
+        or _CIVisibility.enabled
+    ):
+        return
+
+    _CIVisibility.enable(config=ddtrace.config.unittest)
+
+    unittest._datadog_patch = True
+
+    _w = wrapt.wrap_function_wrapper
+
+    _w(unittest, "TextTestResult.addSuccess", add_success_test_wrapper)
+    _w(unittest, "TextTestResult.addFailure", add_failure_test_wrapper)
+    _w(unittest, "TextTestResult.addError", add_failure_test_wrapper)
+    _w(unittest, "TextTestResult.addSkip", add_skip_test_wrapper)
+
+    _w(unittest, "TestCase.run", start_test_wrapper_unittest)
+
+
+def unpatch():
+    if not getattr(unittest, "_datadog_patch", False):
+        return
+
+    _u(unittest.TextTestResult, "addSuccess")
+    _u(unittest.TextTestResult, "addFailure")
+    _u(unittest.TextTestResult, "addError")
+    _u(unittest.TextTestResult, "addSkip")
+    _u(unittest.TestCase, "run")
+
+    unittest._datadog_patch = False
+    _CIVisibility.disable()
+
+
+def _set_test_span_status(args, status):
+    test_item = args[0]
```

Comment:
We are we only checking the args for the status of the test? Can the status of a test be passed in as a keyword argument? If so we should use this helper: https://github.com/DataDog/dd-trace-py/blob/00f9ed9a279b0786f38d3f5c51599cf6337ff053/ddtrace/internal/utils/__init__.py#L17

Reply:
The `args` parameter refers to a tuple of `(test_item, reason)`. I have just changed the variable naming to make it clearer. I do not think it is necessary to use the helper in this case.